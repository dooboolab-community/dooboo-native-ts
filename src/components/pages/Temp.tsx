import {Button, Icon, useTheme} from 'dooboo-ui';
import type {
  RootStackNavigationProps,
  RootStackParamList,
} from '../navigations/RootStack';

import React from 'react';
import type {RouteProp} from '@react-navigation/core';
import styled from '@emotion/native';

const Container = styled.View`
  flex: 1;
  align-self: stretch;

  background-color: ${({theme}) => theme.background};
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

interface Props {
  navigation: RootStackNavigationProps<'Temp'>;
  route: RouteProp<RootStackParamList, 'Temp'>;
}

function Page(props: Props): React.ReactElement {
  const {theme} = useTheme();

  const {
    route: {
      params: {param},
    },
    navigation,
  } = props;

  return (
    <Container>
      <Button
        testID="btn-back"
        onPress={(): void => navigation.goBack()}
        text={param}
        startElement={
          <Icon
            name="chevron-left-light"
            color={theme.textContrast}
            style={{
              marginRight: 4,
            }}
          />
        }
      />
    </Container>
  );
}

export default Page;
