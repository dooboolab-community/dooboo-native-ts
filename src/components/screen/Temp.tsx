import { RootStackNavigationProps, RootStackParamList } from '../../components/navigation/RootStackNavigator';

import Button from '../shared/Button';
import React from 'react';
import { RouteProp } from '@react-navigation/core';
import styled from 'styled-components/native';

const Container = styled.View`
  flex: 1;
  background-color: ${(props): string => props.theme.background};
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

interface Props {
  navigation: RootStackNavigationProps<'Temp'>;
  route: RouteProp<RootStackParamList, 'Temp'>;
}

function Page(props: Props): React.ReactElement {
  const { route: { params: { param } }, navigation } = props;
  return (
    <Container>
      <Button
        testID="btn-back"
        onClick={(): void => navigation.goBack()}
        text={param}
        style={{
          backgroundColor: '#333333',
        }}
      />
    </Container>
  );
}

export default Page;
