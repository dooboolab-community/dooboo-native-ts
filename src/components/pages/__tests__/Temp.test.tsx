import 'react-native';

import {RenderAPI, act, fireEvent, render} from '@testing-library/react-native';
import {createTestElement, createTestProps} from '../../../../test/testUtils';

import {ReactElement} from 'react';
import Temp from '../Temp';

let props: any;
let component: ReactElement;
let testingLib: RenderAPI;

describe('[Temp] render', () => {
  props = createTestProps({
    route: {
      params: {
        param: 'GO BACK',
      },
    },
  });

  component = createTestElement(<Temp {...props} />);

  it('renders without crashing', () => {
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  it('should render [Dark] theme', () => {
    props = createTestProps({
      route: {
        params: {
          param: 'GO BACK',
        },
      },
    });

    component = createTestElement(<Temp {...props} />, 'dark');
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('[Temp] Interaction', () => {
  let renderResult: RenderAPI;

  beforeEach(() => {
    renderResult = render(component);
  });

  it('should simulate [onClick] when button has been clicked', () => {
    const btnInstance = renderResult.getByTestId('btn-back');

    act(() => {
      fireEvent.press(btnInstance);
    });

    expect(props.navigation.goBack).toHaveBeenCalled();
  });
});
