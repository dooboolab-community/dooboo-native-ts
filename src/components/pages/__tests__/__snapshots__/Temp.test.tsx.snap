// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Temp] render renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      activeOpacity={0.6}
      delayPressIn={50}
      onPress={[Function]}
      testID="btn-back"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "#393D7A",
              "borderBottomColor": "#393D7A",
              "borderLeftColor": "#393D7A",
              "borderRightColor": "#393D7A",
              "borderTopColor": "#393D7A",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#D3D8E8",
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#D3D8E8",
                },
                Object {
                  "marginRight": 4,
                },
              ],
              Object {
                "fontFamily": "IcoMoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "#D3D8E8",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          GO BACK
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`[Temp] render should render [Dark] theme 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#232323",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      activeOpacity={0.6}
      delayPressIn={50}
      onPress={[Function]}
      testID="btn-back"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "#8A96DC",
              "borderBottomColor": "#8A96DC",
              "borderLeftColor": "#8A96DC",
              "borderRightColor": "#8A96DC",
              "borderTopColor": "#8A96DC",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "marginRight": 4,
                },
              ],
              Object {
                "fontFamily": "IcoMoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          GO BACK
        </Text>
      </View>
    </View>
  </View>
</View>
`;
